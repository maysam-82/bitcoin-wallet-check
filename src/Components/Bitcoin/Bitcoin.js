import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchBitcoin } from '../../actions/bitcoin/bitcoin';

export class Bitcoin extends Component {
	componentDidMount() {
		this.props.fetchBitcoin();
	}
	calculateBitcoin = () => {
		const { balance, bitcoin } = this.props;
		if (Object.keys(bitcoin).length === 0 || !balance) return '';
		return balance / parseInt(bitcoin.bpi.USD.rate.replace(',', ''));
	};
	render() {
		return <h3>Bitcoin Balance: {this.calculateBitcoin()}</h3>;
	}
}

const mapStateToProps = ({ bitcoin, balance: { balanceValue } }) => {
	return {
		bitcoin,
		balance: balanceValue,
	};
};

export default connect(mapStateToProps, { fetchBitcoin })(Bitcoin);
