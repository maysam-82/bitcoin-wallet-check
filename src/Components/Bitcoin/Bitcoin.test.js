import React from 'react';
import { mount, shallow } from 'enzyme';
import { Bitcoin } from './Bitcoin';

describe('Bitcoin Component', () => {
	const mockFetchBitcoin = jest.fn();
	let props = { balance: 10, bitcoin: {} };
	let wrapper = shallow(<Bitcoin {...props} />);
	it('should render Bitcoin component correctly', () => {
		expect(wrapper.debug()).toMatchSnapshot();
	});
	describe('when Bitcoin mounted', () => {
		beforeEach(() => {
			props.fetchBitcoin = mockFetchBitcoin;
			wrapper = mount(<Bitcoin {...props} />);
		});
		it('should dispatch the `fetchBitcoin` method it receives from props', () => {
			expect(mockFetchBitcoin).toHaveBeenCalled();
		});
	});
	describe('when there are valid bitcoin props', () => {
		beforeEach(() => {
			props = { balance: 15, bitcoin: { bpi: { USD: { rate: '1,500' } } } };
			wrapper = shallow(<Bitcoin {...props} />);
		});
		it('should display the correct bitcoin value', () => {
			expect(wrapper.find('h3').text()).toEqual('Bitcoin Balance: 0.01');
		});
	});
});
