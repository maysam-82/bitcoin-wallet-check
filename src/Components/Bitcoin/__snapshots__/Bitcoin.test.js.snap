// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bitcoin Component should render Bitcoin component correctly 1`] = `
"<h3>
  Bitcoin Balance: 
</h3>"
`;
